lib/spack/spack/installer.py

```diff
@@ -0,0 +1,1554 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+# TODO: Debug erroneous/failing (install) tests: 16 failed, 1 error
+# TODO: Add unit tests
+
+# TODO: Change Installer to process multiple (explicit) package installs
+# TODO:    at once (instead of one)
```

Comment:
Another great catch.  Thank you.  (I should mark these differently so I catch them sooner.)

I will pull these out of the code since they are probably more appropriately in the PR description.

I planned on checking code coverage results and add unit tests accordingly.  Have not yet gotten to that point.

As for changing the installer to process multiple (explicit) package installations at once, IMHO that should be a separate issue and PR from this preliminary work but I will defer to @tgamblin .